src/webots/sound/WbSoundClip.cpp

```diff
@@ -34,6 +34,8 @@ WbSoundClip::WbSoundClip() : mDevice(NULL), mBuffer(0), mSide(0), mBalance(0.0)
 WbSoundClip::~WbSoundClip() {
   if (mBuffer && WbSoundEngine::openAL())
     alDeleteBuffers(1, &mBuffer);
+
+  delete mDevice;
```

Comment:
This is not the responsibility of the WbSoundClip class to handle the cycle of life of the QIODevice object. By the way, this member seems to be useless and could probably be removed.

Reply:
Removed mDevice from WbSoundClip  class.